src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
+        fVar_y = shand_i->mPos.y + local_d0.y + local_e8.y;
+        if (fVar_y < actor->field_CE0) fVar_y = actor->field_CE0;
+        delta_y = fVar_y - shand_i[-1].mPos.y;
+        fVar_z = shand_i->mPos.z - shand_i[-1].mPos.z + d0_scaled.z + local_e8.z;
+        
+        short iVar6;
+        int iVar5 = cM_atan2s(fVar_x,fVar_z);
+        iVar6 = -cM_atan2s(delta_y, std::sqrtf(fVar_x * fVar_x + fVar_z * fVar_z));
+        mDoMtx_YrotS(*calc_mtx, iVar5);
+        mDoMtx_XrotM(*calc_mtx, iVar6);
+        MtxPosition(&local_b8, &cStack_c4);
+        shand_i->mPos = shand_i[-1].mPos + cStack_c4;
+    }
 }
 
 /* 00000BAC-00000C30       .text cut_control3__FP11shand_class */
-void cut_control3(shand_class*) {
-    /* Nonmatching */
+void cut_control3(shand_class* actor) {
+    shand_s* shand_i = actor->field_31C;
+    int counter = 0;
+    for(int i = 20; i != 0; i--, counter++, shand_i++){
+        if(counter < 12){
+            shand_i->u18 = actor->field_304;
+        }
+        else {
+            shand_i->u18 = actor->field_304 * (1.0f - (counter - 10) * 0.08f);
+        }
+    }
 }
 
 /* 00000C30-00000FF0       .text normal__FP11shand_class */
-void normal(shand_class*) {
-    /* Nonmatching */
+void normal(shand_class* actor) {
+    cXyz unused;
+    unused.x = unused.y = 0;
+
+    if(actor->field_318 != 0){
+        dBgS_LinChk local94;
+        cXyz chk_start = actor->current.pos, chk_end = actor->current.pos;
+        chk_start.y += 50.0f;
+        chk_end.y += 4000.0f;
+        local94.Set(&chk_start, &chk_end, actor);
+        if(dComIfG_Bgsp()->LineCross(&local94) != false)
+            actor->field_31C[19].mPos = actor->field_2D4 = actor->field_2C8 = local94.GetCross();
+        actor->field_318--;
+    }
+
+    switch(actor->field_2BA){
+        case 0:
+            if(std::abs(actor->field_31C[19].mPos.y - actor->field_2D4.y) < 10.0f){
+                actor->field_2BA = 1;
+                *actor->field_314 = 2;
+            }
+        case 1:
+            actor->field_2D4 = actor->field_2C8;
+    }
+
+    float y_diff = std::abs(actor->home.pos.y - actor->field_2C8.y);
+    cLib_addCalc2(&actor->field_2F4,   y_diff * (REG14_F(11) + 0.05f), 0.1f, 1.0f);
+    cLib_addCalc2(&actor->field_300, REG14_F(12) + 10.0f, 0.1f, 0.5f);
+    control1(actor);
+    control2(actor);
+    control3(actor);
 }
 
 /* 00001428-00001508       .text cut__FP11shand_class */
-void cut(shand_class*) {
-    /* Nonmatching */
+void cut(shand_class* actor) {
+    cLib_addCalc2(&actor->field_2F4, REG8_F(12) + 20.0f, 0.1f, 0.5f);
+    cut_control(actor);
+    cut_control3(actor);
+    if(actor->field_2BC[0] == 0 && actor->field_30C->health != 0){
+        actor->mState = 0;
+        actor->field_2BA = 0;
+        actor->field_2F0 = 0.0f;
+        actor->field_300 = 0.0f;
+        fopAcM_seStart(actor, JA_SE_OBJ_VINE_S_RECOVER, 0);
+    }
 }
 
 /* 00001508-00001DAC       .text hand_move__FP11shand_class */
-void hand_move(shand_class*) {
-    /* Nonmatching */
```

Comment:
You said three of these functions don't match, so you should keep the nonmatching comments in. You can expand on them to explain what the problem is if you want:
`/* Nonmatching - regalloc */`